### PR TITLE
Feature/gm 12 account info

### PR DESCRIPTION
**Pull request by Morshall2508**

### CODE REVIEW

**Comment by piekaa** [comment]
Would be nice to see my username and avatar, but it can be done in scope of GM-32

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 1]
```diff
@@ -1,18 +1,29 @@
 package pl.piekoszek.gorskimatches.token;
 
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
+import pl.piekoszek.gorskimatches.repository.AccountRepository;
 
+@CrossOrigin
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:7]: why?

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 @RestController
 @RequestMapping("auth")
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:9]: Other endpoints starts with /api, so let's do it here 2

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
 class AccountController {
+    @Autowired
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:11]: do not use field dependency injection, use constructor

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
+
     private EmailService emailService;
 
-    AccountController(EmailService emailService){
+    AccountController(EmailService emailService) {
         this.emailService = emailService;
     }
 
     @PostMapping("email")
     void getEmail(@RequestBody AccountInfo accountInfo) {
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:21]: find better method name ;)

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
         emailService.sendRegistrationLink(accountInfo.getEmail());
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:22]: hmmm it's not only registration, but login as well

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 6]
```diff
@@ -1,18 +1,29 @@
...
     }
+
+    @PostMapping("account")
+    void changeAccountInfo(@RequestBody AccountInfo accountInfo) {
+        accountRepository.save(new AccountInfo(accountInfo.getEmail(), accountInfo.getAccountName(), accountInfo.getAvatar()));
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:27]: is there need to rewrite to `new AccountInfo`? 
just use `accountRepository.save(accountInfo)`

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java [hunk 7]
```diff
@@ -1,14 +1,41 @@
 package pl.piekoszek.gorskimatches.token;
 
+import javax.persistence.*;
+
+@Entity
+@Table(name = "accountInfo")
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:6]: is this annotation required?

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java [hunk 8]
```diff
@@ -1,14 +1,41 @@
...
 public class AccountInfo {
 
+    @Id
+    @Column
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:10]: is this annotation required?

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java [hunk 9]
```diff
@@ -1,14 +1,41 @@
...
     private String email;
+    @Column
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:12]: is this annotation required?

File: src/main/resources/static/account.html [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+<form>
+    <p>Write your username</p><input id="Username">
+    <p>Select avatar</p>
+    <main>
+        <input type="file" accept="image/*" onchange="encodeImageFileAsURL(this)">
+        <img id="output" />
+    </main>
+    <button id="confirm">Confirm changes</button>
+</form>
+<script>
```
Comment (piekaa) [inline src/main/resources/static/account.html:31]: JavaScript code on this page could be improved, but let's do it in scope of GM-32

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 1 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:7]: Good question, I think I was prepping it from some code and thought it was needed to work.

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 2 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:9]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 3 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:11]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 4 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:21]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 5 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:22]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 6 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:27]: right, thanks !

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java (hunk 7 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:6]: not needed

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java (hunk 8 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:10]: not needed

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java (hunk 9 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountInfo.java:12]: not needed

File: src/main/resources/static/account.html (hunk 10 above)
Comment (Morshall2508) [reply src/main/resources/static/account.html:31]: ok!

File: src/main/java/pl/piekoszek/gorskimatches/repository/AccountRepository.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+package pl.piekoszek.gorskimatches.repository;
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+import pl.piekoszek.gorskimatches.token.AccountInfo;
+
+@Component
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/repository/AccountRepository.java:8]: `@Repository` is enough, remove `@Component` ;)

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java [hunk 12]
```diff
@@ -1,18 +1,27 @@
 package pl.piekoszek.gorskimatches.token;
 
 import org.springframework.web.bind.annotation.*;
+import pl.piekoszek.gorskimatches.repository.AccountRepository;
 
 @RestController
-@RequestMapping("auth")
+@RequestMapping("api")
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:7]: auth is also fine, go with api/auth ;)

File: src/main/java/pl/piekoszek/gorskimatches/token/EmailService.java [hunk 13]
```diff
@@ -18,7 +17,7 @@ public class EmailService {
         this.server = server;
     }
 
-    public void sendRegistrationLink(String to) {
+    public void sendAccountActivationLink(String to) {
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/token/EmailService.java:20]: `sendRegistrationOrLoginLink`?

**Review by piekaa — CHANGES_REQUESTED** [review]
few minor things

File: src/main/java/pl/piekoszek/gorskimatches/repository/AccountRepository.java (hunk 11 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/repository/AccountRepository.java:8]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java (hunk 12 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/AccountController.java:7]: ok

File: src/main/java/pl/piekoszek/gorskimatches/token/EmailService.java (hunk 13 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/token/EmailService.java:20]: ok :)

**Review by piekaa — APPROVED** [review]

**Pull request merged**